File: eRent/eRent.Services/Database/ViewingAppointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eRent.Services.Database
{
    public class ViewingAppointment
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int PropertyId { get; set; }

        [Required]
        public int TenantId { get; set; } // FK to User (the person requesting viewing)

        [Required]
        public DateTime AppointmentDate { get; set; } // Date + start time

        [Required]
        public DateTime EndTime { get; set; } // AppointmentDate + 2 hours

        /// <summary>
        /// 0 = Pending, 1 = Approved, 2 = Rejected, 3 = Cancelled, 4 = Completed
        /// </summary>
        public int Status { get; set; } = 0;

        [MaxLength(500)]
        public string? TenantNote { get; set; }

        [MaxLength(500)]
        public string? LandlordNote { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("PropertyId")]
        public Property Property { get; set; } = null!;

        [ForeignKey("TenantId")]
        public User Tenant { get; set; } = null!;
    }
}

File: Startupba/UI/erent_desktop/lib/model/viewing_appointment.dart
import 'package:json_annotation/json_annotation.dart';

part 'viewing_appointment.g.dart';

@JsonSerializable()
class ViewingAppointment {
  final int id;
  final int propertyId;
  final String propertyTitle;
  final String propertyAddress;
  final int tenantId;
  final String tenantName;
  final int landlordId;
  final DateTime appointmentDate;
  final DateTime endTime;
  final int status;
  final String statusName;
  final String? tenantNote;
  final String? landlordNote;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const ViewingAppointment({
    this.id = 0,
    this.propertyId = 0,
    this.propertyTitle = '',
    this.propertyAddress = '',
    this.tenantId = 0,
    this.tenantName = '',
    this.landlordId = 0,
    required this.appointmentDate,
    required this.endTime,
    this.status = 0,
    this.statusName = '',
    this.tenantNote,
    this.landlordNote,
    required this.createdAt,
    this.updatedAt,
  });

  factory ViewingAppointment.fromJson(Map<String, dynamic> json) =>
      _$ViewingAppointmentFromJson(json);
  Map<String, dynamic> toJson() => _$ViewingAppointmentToJson(this);
}
